### DISCUSSION TITLE
GitHub Copilot not showing suggestions in jupyter notebooks (PyCharm/DataSpell)

### DISCUSSION THREAD

**Discussion by Deadeye111** [community/community#17185] [Copilot Conversations]
Hi
I am using GitHub Copilot in PyCharm and DataSpell. The code suggestions work fine in .py files but don't work in Jupyter Notebook files anymore. I currently have the newest available version installed (1.1.22.1530). 
Sometimes there are following warnings in the idea.log file: 
- WARN - efaultCopilotCompletionService - fetchCompletions: GitHub Copilot token secret not defined or invalid: null
- WARN - efaultCopilotCompletionService - Temporary file is invalid or physical: Jupyter Notebook 

I don't know if these warnings are related to the problem.
For both PyCharm and DataSpell I have version 2021.3.3 installed. Reinstalling the plugin or logging off and on didn't work. Does anybody else have the same issue?

Other than that, I'm really excited about copilot so far and I like to experiment with it.

**Comment by jansorg** [comment]
Thanks for reporting!
Could you perhaps post a screenshot showing the editor where you'd expect suggestions by GitHub Copilot?

**Reply by Deadeye111** [reply]
Sure, for example:
![image](https://user-images.githubusercontent.com/61310625/170260963-ff95acf1-7cc7-417c-8e01-13af6e3cdab3.png)

**Reply by alibagheri7** [reply]
I am facing the same problem,

**Reply by federicomoreno613** [reply]
The same problem here!

**Comment by gregpriday** [comment]
Same issue here. Copilot worked flawlessly in PyCharm with Notebooks since I started using it a few months ago. It stopped working a few days ago. Copilot still works perfectly in standard python files.

**Reply by Deadeye111** [reply]
Last time it worked for me in Notebooks was in version 1.1.16.1219 I believe.

**Reply by gregpriday** [reply]
This started working for me again yesterday. After I switched to a trial of the recently released full-paid version of Github Copilot.

**Reply by stevenmortier** [reply]
For me, it still does not work. I am on the student plan now. Did you reinstall the plugin?

**Reply by sadatian** [reply]
Two months later, and a fully paid copilot account and it still not working on notebooks.......

**Reply by kaiwenShen** [reply]
just dont understand how they took so much time in solving what appears to be a compatibility issue

**Reply by gregpriday** [reply]
As a follow-up, I just set up a new Mac Studio. Copilot isn't working in Notebooks on the new Mac Studio, while it's still working on my old Macbook Pro. Both are running the latest versions of PyCharm and the Copilot plugin, both are M1 processors. If I can figure out what the difference between the 2 setups is, I'll post a follow-up. For now, all I can add is that this is possibly configuration related.

**Comment by denmj** [comment]
Same problem, works fine with .py in PyCharm, but stopped working with notebooks in DataSpell.

**Comment by LaviBenshimol** [comment]
Same problem here, too, working with PyCharm and DataSpell for a few months.
Currently not working on DataSpell (working only with .pynb).

**Reply by LaviBenshimol** [reply]
Update: reinstall DataSpell fixed the issue

**Reply by pj-mathematician** [reply]
Reinstalling DataSpell fixes the issue completely?

**Comment by yuvrajiro** [comment]
Same Problem

**Comment by stevenmortier** [comment]
I have the same problem. Downgrading fixed this issue, but since then it has forced me to update to the newest version, which again broke it..

**Comment by figgarden** [comment]
I'm having the same problem on DataSpell 2021.3.1

**Comment by josumsc** [comment]
Same problem here on DataSpell 2021.3.1

**Comment by schauppi** [comment]
Same Problem on DataSpell 2021.3.2

**Reply by jenakal** [reply]
Found any solution? Can you share your feedback? Thanks in advance[.](https://www.azzmedical.com/)

**Comment by AndPuQing** [comment]
Some Problem on Pycharm Professional 2022.2 EAP

**Comment by mjwsolo** [comment]
Facing the same issue on VS Code!

**Reply by MAFiA303** [reply]
I'm facing this issue with VS code 1.71.0

**Reply by Guido-Hwang** [reply]
same issue

**Comment by BiomedicLJZ** [comment]
Facing the same problem on DataSpell 2022.2

**Reply by LirenSong11** [reply]
rolling back to 2022.1.4 solves it
download from here
https://www.jetbrains.com/dataspell/download/other.html

**Reply by chmedinap** [reply]
> rolling back to 2022.1.4 solves it download from here https://www.jetbrains.com/dataspell/download/other.html

thank you!

**Comment by bryzgalovdm** [comment]
Me as well, dataspell Build #DS-222.3345.132

**Comment by Xyndra** [comment]
Yup Same problem here

**Comment by kaiwenShen** [comment]
me too on pycharm 2022.2

**Comment by ellieLitwack** [comment]
Having the same problem on both PyCharm and DataSpell.

**Reply by janus78** [reply]
The same is happening to me in pycharm professional 2022.2 and dataspell 2022.2, github copilot stopped working, I have reinstalled it but is not working.

**Comment by LirenSong11** [comment]
rolling back to 2022.1.4 solves it
download from here
https://www.jetbrains.com/dataspell/download/other.html

**Reply by NoahVl** [reply]
This fixed it! Thank you :)

**Comment by tandrew468** [comment]
PyCharm only .py files worked. Version 2022.2.1
Rolling back to 2022.1.4 seems to have fix the issue with it not working in .ipynb.
If you have the [toolbox](https://www.jetbrains.com/toolbox-app/) download other versions from there.

**Comment by ivanstepanovftw** [comment]
I paid $20 for nothing

**Reply by jaded0** [reply]
Yeah I'm glad I got the student plan because otherwise I'd be looking for my money back.

**Comment by AleksandrovichK** [comment]
Fucking hell, Github!
Why would I pay for something, that doesn't have a basic support?
For something that doesn't work in a fucking basic use-case. This is outrageous.

**Reply by tandrew468** [reply]
https://github.com/orgs/community/discussions/17185#discussioncomment-3408949

**Reply by AleksandrovichK** [reply]
@tandrew468 Rollback is not a solution in a long term, because JetBrains keeps releasing new versions that include important **security updates.**

**Comment by azenar** [comment]
So, there's no update regarding this issue? (DataSpell 2022.2.1 or any versions that face the same problem) 
It's been months already. So the only solution that works right now is rolling back?

**Comment by alansastre** [comment]
Same problem with PyCharm Professional 2022.2.1

**Reply by tandrew468** [reply]
https://github.com/orgs/community/discussions/17185#discussioncomment-3408949

**Comment by trainma** [comment]
Same problem with PyCharm Professional 2022.2.2. When to fix it?

**Reply by trainma** [reply]
copilot can work with jupyter in vscode.

**Comment by spirosbax** [comment]
same problem with dataspell 2022.2.2

**Comment by Vladykart** [comment]
Same problem with PyCharm Professional 2022.2.1

**Reply by Vladykart** [reply]
not fixed after update to copilot 1.1.31.1946 
with PyCharm Professional 2022.2.2
I pay for this tool to work with jupiter notebooks

**Comment by aljungberg** [comment]
I can confirm this has not been resolved as of PyCharm Professional 2022.2.2 (222.4167.33) with GitHub Copilot 1.1.32.1956. Continues to work with regular `.py` but not with Jupyter notebooks.

(On a related note, I wish the Copilot Change Notes weren't just an afterthought at the bottom of the plugin description with a nebulous "Various bug fixes" permanently tacked on. If it said *which* bugs were fixed one wouldn't have to keep upgrading Pycharm just to see if this has now been fixed and then downgrade again if not.)

**Comment by stefanpricopie** [comment]
It doesn't work on *.ipynb files in PyCharm 2022.2.2 (Professional Edition), but it does work on *.ipynb files when using VScode.

**Comment by TransMux** [comment]
This is not fixed for the latest version of pycharm, but according to this [issue](https://youtrack.jetbrains.com/issue/DS-3756/Github-Copilot-does-not-work-in-Dataspell-with-Jupyter-notebooks#focus=Comments-27-6398892.0-0), falling back to the version below will work.

> PyCharm 2022.1.4 (Professional Edition)
Build #PY-221.6008.17, built on July 20, 2022
Runtime version: 11.0.15+10-b2043.56 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 750M
Cores: 8
Non-Bundled Plugins:
    com.github.copilot (1.1.30.1887)
    
You can download here: https://www.jetbrains.com/pycharm/download/other.html

I tried, and it just worked. But i'm still looking for the fix for the latest version.

**Comment by leleogere** [comment]
Rolling back to 2022.1.4 fixes the issue, but that's not a long-term solution, we expect copilot to work with the latest releases for obvious security reasons.

**Comment by trainma** [comment]
Hi，2022.2.3 also not work when to fix? Fvck Microsoft

**Comment by pantlinardatos** [comment]
Same problem with PyCharm Professional 2022.2.3

**Comment by jaded0** [comment]
Still a problem. After 6 months, is this ever going to be fixed?

**Comment by Arengard** [comment]
it could be decades before they get it to work

**Comment by mf093087** [comment]
same for me!!

**Comment by amanape** [comment]
How disappointing...

**Comment by yixunx** [comment]
Same problem using the Python plugin in IntelliJ Ultimate 2022.2.2. I can switch to PyCharm, but looks like PyCharm doesn't work either according to previous comments.

**Comment by Ethan-Loo** [comment]
Are there any co-pilot devs that work across the aisle with JetBrains devs? 
There are multiple threads in the YouTrack JetBrains forums about this specific loss of functionality with jupyter notebooks and/or .ipynb files in particular such as this one for DataSpell: https://youtrack.jetbrains.com/issue/DS-3756

I think some collaboration is needed between teams to sort this particular issue out because there doesn't seem to be a clear way to identify the exact source of the problem on the user end. I feel like rolling back to a previous version of JetBrains software is a cop-out since that means relinquishing any current and future updates in both features and security just to keep jupyter functionality.

**Comment by shacharhelmer** [comment]
Just updated to `1.1.35.2063` and the change notes say: _Improve compatibility with 2022.3_.
So I went and updated data spell to 2022.3 EAP and guess what? Exactly! There is not compatibility to improve in .ipynb at all!!

I suggest to anyone facing this issue - give up waiting for this to be fixed, switch to tabnine. It works in .ipynb files in all versions.
I am in no conflict of interests here, just a tired customer who's fed up with being ignored.

**Reply by Snimm** [reply]
There is no student version of tabnine.

**Reply by shacharhelmer** [reply]
@Snimm, Problematic I agree. Copilot would be ideal.
However there is a free more limited version.

**Comment by 0xzhouchenyu** [comment]
I have the same question. Don't you really want to fix it?

**Comment by mf093087** [comment]
Stick to Pycharm 2022.1.x version, that is the current solution

**Comment by parsanoori** [comment]
Can someone give a tl;dr?
I have the problem too.
What's the solution?

**Reply by Aivean** [reply]
> Stick to Pycharm 2022.1.x version, that is the current solution

**Comment by rickythefox** [comment]
Seems like the JetBrains team has started working on a solution: https://youtrack.jetbrains.com/issue/DS-3756/Github-Copilot-does-not-work-in-Dataspell-with-Jupyter-notebooks

**Comment by StavC** [comment]
not working for me aswell at the newest versions.

I will try to downgrade my version.

**Comment by Xyndra** [comment]
now working but you need to trust jupyter

**Reply by RafailFridman** [reply]
In which version does it work for you?

**Reply by StavC** [reply]
Try Pycharm 2022.1.4 Pro, works for me.

**Comment by StavC** [comment]
It works for me at Pycharm 2022.1.4 Pro , downgrade your pycharm versions and it will work for you guys.

it's probably bugged in the newer versions.

**Reply by shacharhelmer** [reply]
@StavC This was mentioned here before, but is a not feasible solution given that security updates and new features are missed when downgrading...

**Reply by StavC** [reply]
> @StavC This was mentioned here before, but is a not feasible solution given that security updates and new features are missed when downgrading...

True, Unfortunately I couldn't find a better temporary fix.

**Comment by sschonss** [comment]
same problem with dataspeel 2022.3

**Reply by sschonss** [reply]
works now!

**Comment by iiDeSTRoYeR** [comment]
Same problem with 2022.2.4.
Any ETA on the fix?

**Comment by 1nicolaspetermann** [comment]
same Problem. Please fix it

**Comment by RebornX10** [comment]
Same issue here, 


```
DataSpell 2022.3
Build #DS-223.7571.211, built on December 1, 2022
Licensed to XXXXXXXX
Subscription is active until June 9, 2023.
For educational use only.
Runtime version: 17.0.5+1-b653.14 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.10.0-19-amd64
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 8
Registry:
    ea.indicator.blinking.timeout=0
    python.scientific.always_use_sci_view=true
    python.scientific.show_help_window=false

Non-Bundled Plugins:
    com.tabnine.TabNine (0.8.2)
    com.smallcloud.codify (0.1.3)
    com.github.copilot (1.1.37.2188)
    net.seesharpsoft.intellij.plugins.csv (3.0.1-223)
    izhangzhihao.rainbow.brackets (2022.3.5)

Current Desktop: KDE
```
Switched to another completion plugin while waiting for it to be fixed...

**Comment by Alhern** [comment]
Guess what, it still doesn't work with PyCharm 2022.3.

**Reply by NashTools** [reply]
same here

**Comment by alansastre** [comment]
6 months after and GitHub Copilot is still not working in Jupyter Notebooks, even in last version PyCharm Professional 2022.3.

**Comment by datatrickster** [comment]
Same thing! Fix it please

**Comment by Aivean** [comment]
Hey folks, it appears [some progress has been made](https://youtrack.jetbrains.com/issue/DS-3756#focus=Comments-27-6754283.0-0):
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/2865203/211130011-b385b1af-3cbd-4c26-87ea-2c64b34b3964.png">

**Reply by RafailFridman** [reply]
Can anyone please tell if it works now in 2022.3?

**Reply by darkydash** [reply]
Doesn't work in 2022.2.4 sadly. Fine for .py, broken for .ipynb

**Reply by Ieremie** [reply]
Same problem here!

**Reply by Lior-Baruch** [reply]
have this problem for a while, just updated the copilot and it is still not working :(

**Reply by alexander-friedrich** [reply]
Same problem here! I have been waiting for a fix for a while now :(

**Reply by sherlocked27** [reply]
Best solution for now is to downgrade to 2022.1.4

**Comment by aaSchcolnik** [comment]
I was working with DataSpell for the  past few days, github copilot was working properly on jupyter notebooks, I updated DataSpell today while updating my plugins and Copilot stopped working in jupyter notebooks. Unfortunately i never checked what version i was working with, i tried downgrading to 2022.1.4 but it didn't fix the issue.. bummer since i prefer dataspell over vscode

**Comment by pantlinardatos** [comment]
FIXED. With the new copilot update. (Pycharm pro 2022.3.2)

**Reply by Aivean** [reply]
I confirm, works with IntelliJ IDEA 2022.3.2 (Build #IU-223.8617.56) and Copilot plugin 1.2.1.2347

<img width="349" alt="image" src="https://user-images.githubusercontent.com/2865203/215278292-7efb519b-c086-4989-88a6-74afbb3e3ddd.png">

Thank you, github and intellij idea folks!

**Reply by nix1** [reply]
Ooh finally 🎉  I confirm, works with DataSpell 2022.3.2 (Build #DS-223.8617.58, Copilot 1.2.1.2347)

**Comment by stefanpricopie** [comment]
Can confirm as well for PyCharm. Works with PyCharm 2022.2.4 and Copilot 1.2.1.2347. Thank you!

**Comment by Ieremie** [comment]
It now works, but only in the suggestion panel. In line suggestions do not appear as before.

**Comment by mirlan-irokez** [comment]
it now works for me in DS-223.8617.58 and GH-Copilot 1.2.3.2385

**Comment by bukowa** [comment]
Not working for me in Build #PY-231.8109.197, built on March 29, 2023

**Reply by amanpreet692** [reply]
Yep, not working for me either with the latest Professional build

**Reply by bukowa** [reply]
Working for me now, Build #PY-231.8109.197, built on March 29, 2023

**Reply by Frightera** [reply]
Try switching into old UI if you are using the new one (#PY-231.8109.197)

**Comment by JalinWang** [comment]
There is a discussion on youtrack:
 https://youtrack.jetbrains.com/issue/DS-3756/Github-Copilot-does-not-work-in-DataSpell-with-Jupyter-notebooks#focus=Comments-27-7330856.0-0

**Comment by dbalabka** [comment]
It is still a problem. Steps to reproduce:
1. Create a project located in WSL FS
2. Open any notebook 
3. Try to get any suggestions and interact with chat, you will see an error

More details here: https://youtrack.jetbrains.com/issue/PY-66264/Github-Copilot-does-not-work-with-Jupyter-notebooks-if-WSL-is-used
